ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/internal/privacy/methods/priv/PrivGetTransactionReceipt.java

```diff
@@ -88,24 +89,48 @@ public JsonRpcResponse response(final JsonRpcRequestContext requestContext) {
     final Transaction pmtTransaction =
         blockBody.getTransactions().get(pmtLocation.getTransactionIndex());
 
-    final Hash blockhash = pmtLocation.getBlockHash();
-    final long blockNumber = blockchain.getBlockchain().getBlockHeader(blockhash).get().getNumber();
+    final Hash blockHash = pmtLocation.getBlockHash();
+    final long blockNumber = blockchain.getBlockchain().getBlockHeader(blockHash).get().getNumber();
 
     final PrivateTransaction privateTransaction;
     final String privacyGroupId;
     try {
       final ReceiveResponse receiveResponse =
           privacyController.retrieveTransaction(
-              pmtTransaction.getPayload().toBase64String(),
+              pmtTransaction.getPayload().slice(0, 32).toBase64String(),
               enclavePublicKeyProvider.getEnclaveKey(requestContext.getUser()));
       LOG.trace("Received private transaction information");
 
       final BytesValueRLPInput input =
           new BytesValueRLPInput(
               Bytes.fromBase64String(new String(receiveResponse.getPayload(), UTF_8)), false);
-
-      privateTransaction = PrivateTransaction.readFrom(input);
-      privacyGroupId = receiveResponse.getPrivacyGroupId();
+      input.enterList();
+      if (input.nextIsList()) {
+        privateTransaction = PrivateTransaction.readFrom(input);
+        input.leaveListLenient();
+      } else {
+        input.reset();
+        privateTransaction = PrivateTransaction.readFrom(input);
+      }
+
+      if (privateTransaction.getPrivacyGroupId().isPresent()) {
```

Comment:
The whole logic to find out what the privacyGroupId is looks a bit too complicated. If there is a privacyGroupId in the private transaction, we could just use that, if not just use the privacyGroupId returned in the receiveResponse ...

Reply:
fixed